homeassistant/components/roomba/irobot_base.py

```diff
@@ -192,7 +193,13 @@ def extra_state_attributes(self):
         if self.state == STATE_CLEANING:
             # Get clean mission status
             mission_state = state.get("cleanMissionStatus", {})
-            cleaning_time = mission_state.get("mssnM")
```

Comment:
This would be easier to read if you check if mission_state is None, then this: 
```python
if not mission_state:
  state_attrs[ATTR_CLEANING_TIME] = 0
  state_attrs[ATTR_CLEANED_AREA] = 0
else:
  cleaning_time = mission_state.get("mssnM", None)
  if cleaning_time:
    state_attrs[ATTR_CLEANING_TIME] = cleaning_time
  else:
    start_time = mission_state.get("mssnStrtTm", None)
    if not start_time:
       state_attrs[ATTR_CLEANING_TIME] = 0
    else:
      state_attrs[ATTR_CLEANING_TIME] = (dt_util.as_timestamp(dt_util.utcnow()) - start_time                    ) // 60
```
even better with a seperate function, so you can just return instead of using else (and a nice use case for the walrus!)!
something like this:
```python
if self.state == STATE_CLEANING:
  state_attrs[ATTR_CLEANING_TIME] = self.get_cleaning_time(state)

def get_cleaning_time(self, state) -> int:
  if not (mission_state := state.get("cleanMissionStatus", None)):
    return 0
  if cleaning_time := mission_state.get("mssnM", None):
    return cleaning_time
  if not (start_time := mission_state.get("mssnStrtTm", None)):
    return = (dt_util.as_timestamp(dt_util.utcnow()) - start_time) // 60
  return 0
```

Reply:
I'll look at that, but my initial thoughts are:

1. `cleanMissionStatus` does still contain other information, so it's unlikely to be `None`.

2. I only have one robot to test with (Roomba 960), and I'm not sure of all of the models have removed the missing statistics.

3. I've never used the walrus 😅

Reply:
1. doesn't change the fact that you need to check for it anyway and then move on quickly
2. not sure what you mean with the missing statistic!
3. I haven't had a good case for it myself, but this one is nice!

Reply:
For 2, I'm saying that there may be models which *do* still report `mssnM` in their response.

It seems that, at least for my model, neither the cleaning time nor area are reported anymore. I haven't figured out a way to get the area yet, but this fixed the cleaning time, in the instance where it never changes from 0.

Reply:
I've just made the requested changes :+1: Makes sense to me 😁

Reply:
Looks better, one small refinement now is to do something similar with the cleaning_area, potentially in the same structure or even by using a tuple as the return from a single function with both time and area, so that you only have to read the mission_state once!

Reply:
I'm still looking into this, but not very successfully. Alas, it seems that nothing useful in calculating the cleaned area is included in the response (from my robot, at least) anymore. The map gets generated in the iRobot app, along with cleaning time and cleaned area, but I presume it is now being handled server-side somehow.